DataFormats/MuonReco/interface/MuonEnergy.h

```diff
@@ -49,13 +51,36 @@ namespace reco {
        /// DetId of the central HCAL tower with smallest depth
        DetId hcal_id;
        
+       ///
+       /// RecHits
+       ///
+       /// crossed HCAL rechits
+       const std::vector<HBHERecHit>* getHadRecHits() const
+       {
+	 if (hadRecHitsValid) return &hadRecHits;
+	 return nullptr;
+       }
+		 
+       void setHadRecHits(const std::vector<const HBHERecHit*>& rechits)
+       {
+	 hadRecHitsValid = true;
+	 hadRecHits.clear();
+	 for (auto hit: rechits)
+	   hadRecHits.push_back(*hit);
+       }
+
        MuonEnergy():
        tower(0), towerS9(0),
        em(0), emS9(0), emS25(0), emMax(0),
        had(0), hadS9(0), hadMax(0),
-	   ho(0), hoS9(0), ecal_time(0), ecal_timeError(0), hcal_time(0), hcal_timeError(0)
+	 ho(0), hoS9(0), ecal_time(0), ecal_timeError(0), hcal_time(0), hcal_timeError(0),
+	 hadRecHitsValid(false)
 	 { }
-       
+
+    private:

```

Comment:
this is becoming too complex to stay in a struct.
The MuonEnergy should be moved to be a class or the access simplified back to a plain struct (to leave simply trivial access to the data members).

Reply:
will drop it